src/Libraries/RevitNodes/Elements/Views/ScheduleView.cs

```diff
@@ -0,0 +1,256 @@
+            this.InternalUniqueId = view.UniqueId;
+        }
+
+        #endregion
+
+        #region Private helper methods
+
+        private static Autodesk.Revit.DB.ViewSchedule CreateViewSchedule(Category category, string name, ScheduleType type)
+        {
+            Autodesk.Revit.DB.Document doc = DocumentManager.Instance.CurrentDBDocument;
+            TransactionManager.Instance.EnsureInTransaction(doc);
+
+            Autodesk.Revit.DB.ViewSchedule viewSchedule = null;
+            switch (type)
+            {
+                case ScheduleType.KeySchedule:
+                    viewSchedule = Autodesk.Revit.DB.ViewSchedule.CreateKeySchedule(doc, new Autodesk.Revit.DB.ElementId(category.Id));
+                    viewSchedule.Name = name;
+                    break;
+            }
+
+            TransactionManager.Instance.TransactionTaskDone();
+
+            return viewSchedule;
+        }
+
+        #endregion
+
+        #region Public static constructors
+
+        /// <summary>
+        ///     Create Schedule by Category, Type and Name.
+        /// </summary>
+        /// <param name="category"></param>
+        /// <param name="name"></param>
+        /// <param name="scheduleType"></param>
+        /// <returns name="scheduleView">Schedule</returns>
+        public static ScheduleView CreateSchedule(Category category, string name, string scheduleType)
+        {
+            if (category == null || name == null)
+            {
+                throw new System.ArgumentNullException("Category or Name");
+            }
+
+            ScheduleType t = (ScheduleType)System.Enum.Parse(typeof(ScheduleType), scheduleType);
+            return new ScheduleView(category, name, t);
+        }
+
+        ///// <summary>
+        /////     Remove Schedule Field from Schedule.
+        ///// </summary>
+        ///// <param name="field"></param>
+        ///// <returns name="View">View Schedule</returns>
+        //public ScheduleView RemoveField(Revit.Schedule.ScheduleField field)
+        //{
+        //    if (this.Fields.Any(x => x.Name == field.Name))
+        //    {
+        //        TransactionManager.Instance.EnsureInTransaction(DocumentManager.Instance.CurrentDBDocument);
+        //        this.InternalViewSchedule.Definition.RemoveField(field.InternalScheduleField.FieldId);
+        //        TransactionManager.Instance.TransactionTaskDone();
+        //    }
+
+        //    return this;
+        //}
+
+        /// <summary>
+        ///     Add Field.
+        /// </summary>
+        /// <param name="field"></param>
+        /// <returns></returns>
+        public ScheduleView AddField(Revit.Schedule.SchedulableField field)
+        {
+            if (!this.Fields.Any(x => x.Name == field.Name))
+            {
+                TransactionManager.Instance.EnsureInTransaction(DocumentManager.Instance.CurrentDBDocument);
+                this.InternalViewSchedule.Definition.AddField(field.InternalSchedulableField);
```

Comment:
for Key Schedules there cannot be two fields (field is an equivalent of parameter/schedule column) with the same name. checking for name is good enough as that will prevent an error.